simulator/infra/macro.h

```diff
@@ -16,6 +16,7 @@
 #include <limits>
 #include <type_traits>
 #include <cassert>
+#include <array>
```

Comment:
Please sort headers in alphabetical order. While you are here, please remove `cassert` which is not used in that file.